tests/model_selection/test_hyperband.py

```diff
@@ -0,0 +1,385 @@
+import math
+from collections import defaultdict
+
+import dask.array as da
+import numpy as np
+import pytest
+import scipy.stats
+from distributed.utils_test import cluster, gen_cluster, loop  # noqa: F401
+from sklearn.linear_model import SGDClassifier
+
+from dask_ml.datasets import make_classification
+from dask_ml.model_selection import (
+    HyperbandSearchCV,
+    IncrementalSearchCV,
+    SuccessiveHalvingSearchCV,
+)
+from dask_ml.model_selection._hyperband import _get_hyperband_params
+from dask_ml.utils import ConstantFunction
+from dask_ml.wrappers import Incremental
+
+
+@pytest.mark.parametrize(
+    "array_type, library, max_iter",
+    [
+        ("dask.array", "dask-ml", 9),
+        ("numpy", "sklearn", 9),
+        ("numpy", "ConstantFunction", 15),
+        ("numpy", "ConstantFunction", 20),
+    ],
+)
+def test_basic(array_type, library, max_iter):
+    @gen_cluster(client=True, timeout=5000)
+    def _test_basic(c, s, a, b):
+        n, d = (50, 2)
+
+        rng = da.random.RandomState(42)
+
+        # create observations we know linear models can fit
+        X = rng.normal(size=(n, d), chunks=n // 2)
+        coef_star = rng.uniform(size=d, chunks=d)
+        y = da.sign(X.dot(coef_star))
+
+        if array_type == "numpy":
+            X = yield c.compute(X)
+            y = yield c.compute(y)
+
+        params = {
+            "loss": ["hinge", "log", "modified_huber", "squared_hinge", "perceptron"],
+            "average": [True, False],
+            "learning_rate": ["constant", "invscaling", "optimal"],
+            "eta0": np.logspace(-2, 0, num=1000),
+        }
+        model = SGDClassifier(
+            tol=-np.inf, penalty="elasticnet", random_state=42, eta0=0.1
+        )
+        if library == "dask-ml":
+            model = Incremental(model)
+            params = {"estimator__" + k: v for k, v in params.items()}
+        elif library == "ConstantFunction":
+            model = ConstantFunction()
+            params = {"value": np.linspace(0, 1, num=1000)}
+
+        search = HyperbandSearchCV(model, params, max_iter=max_iter, random_state=42)
+        classes = c.compute(da.unique(y))
+        yield search.fit(X, y, classes=classes)
+
+        if library == "ConstantFunction":
+            score = search.best_estimator_.score(X, y)
+            assert score == search.score(X, y)
+            assert 0 <= score <= 1
+            assert score == search.best_score_
+            assert score == max(
+                hist[-1]["score"] for hist in search.model_history_.values()
+            )
+        elif library == "sklearn":
+            score = search.best_estimator_.score(X, y)
+            assert score == search.score(X, y)
+            assert 0 <= score <= 1
+            assert abs(score - search.best_score_) < 0.1
```

Comment:
Why do assertions for `library='dask-ml`?

It'd be nice if the assertions that apply to all the underlying estimators were done outside of this if statement. Perhaps things like `best_estimator_.score` can't be done for Incremental, though I suspect that should be possible.

Reply:
Thanks. This is resolved now. All `assert` statements are done outside the if-statement.